File: Data Structures & Algorithms/longest-palindromic-substring/submission-4.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        finalstr = ""

        def expandFromCenter(left: int,right: int,s: str) -> str:
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1
                right += 1
            return s[left+1:right]

        for i in range(0,len(s)):
            str1 = expandFromCenter(i,i,s)
            str2 = expandFromCenter(i,i+1,s)
            finalstr = max(finalstr,str1,str2,key = len)
        return finalstr

File: Data Structures & Algorithms/palindromic-substrings/submission-4.java
class Solution {
    public int countSubstrings(String s) {
        int count = 0;

        for (int i = 0;i<s.length();i++){
            int c1 = expandFromCenter(i,i,s);
            int c2 = expandFromCenter(i,i+1,s);
            count+=c1 +c2;
        }

        return count;
    }

    private int expandFromCenter(int left,int right,String s){
        int count = 0;
        while(left >= 0  && right <s.length() && s.charAt(left) == s.charAt(right)){
            count++;
            left--;
            right++;
        }
        return count;
    }
}
